frontend/src/components/indices/IndicesValidate.js

```diff
@@ -0,0 +1,260 @@
+import React, {useEffect, useState} from "react";
+import {connect} from "react-redux";
+import {useHistory} from "react-router-dom";
+import {
+  Alert,
+  Button,
+  Form,
+  Input,
+  InputNumber,
+  Select,
+  Cascader,
+} from "antd";
+const {Option} = Select
+
+import AppPageHeader from "../AppPageHeader";
+import PageContent from "../PageContent";
+import * as Options from "../../utils/options";
+import api, {withToken} from "../../utils/api";
+import {validate, list} from "../../modules/indices/actions";
+import {requiredRules} from "../../constants";
+
+// API functions
+const listSets = (token, options) =>
+  withToken(token, api.indices.listSets)(options).then(res => res.data)
+
+const listInstrumentTypes = (token, options) =>
+  withToken(token, api.instruments.listTypes)(options).then(res => res.data)
+
+
+const mapStateToProps = state => ({
+  token: state.auth.tokens.access,
+  indicesTotalCount: state.indices.totalCount,
+  isFetching: state.indices.isFetching,
+});
+
+const actionCreators = {list, validate};
+
+const IndicesValidate = ({token, indicesTotalCount, isFetching, list, validate}) => {
+  const history = useHistory();
+
+  /*
+   * State management
+   */
+
+  const [instrumentTypes, setInstrumentTypes] = useState([]);
+  const [indicesBySet, setIndicesBySet] = useState([]);
+  const [validationLoading, setValidationLoading] = useState(false)
+
+  useEffect(() => {
+    //List instrument
+    listInstrumentTypes(token).then(instrumentTypes => {
+      setInstrumentTypes(instrumentTypes.map(Options.renderInstrumentType))
+    })
+    //List sets and initialize the cascader options
+    listSets(token, {}).then(sets => {
+      sets.map(set => {
+        setIndicesBySet((currentState) => [ ...currentState,
+        {
+          label: set.name,
+          value: set.id,
+          isLeaf: false,
+        }])
+      })
+    })
+  }, [])
+
+  const loadData = (selectedOptions) => {
+    const targetOption = selectedOptions[selectedOptions.length - 1]
+    const setID = targetOption.value
+    const query = {"index_set__id__in": setID}
+    targetOption.loading = true
+
+    // load options lazily
+    setTimeout(() => {
+      list({...query}).then(response => {
+        const indices = response.results
+        //concatenate existing indices to the retrieved ones
+        targetOption.children = indices?.map(index => {
+          return {
+            label: index.name,
+            value: index.id,
+            children: [],
+          }
+        })
+        targetOption.loading = false;
+        setIndicesBySet([...indicesBySet])
+      })
+    }, 1000);
```

Comment:
Is this large enough ? 1000 ms may run out quickly if the server is processing other request at the same time (or with a very large index set).

Reply:
I was actually thinking about decreasing the value. Waiting 1 second before making the request seemed useless, but now I am thinking about leaving it like that

Reply:
yeah well default is to execute right away ... if you want to reduce delays for the user that could be nice ;).